### DISCUSSION TITLE
Problema com couldn't find "libflutter.so"

### DISCUSSION THREAD

**Discussion by guilhermerp-gh** [Flutterando/forum#183] [Perguntas e Respostas] [answered]
Boa tarde pessoal. Tenho um aplicativo que será feita a primeira publicação e colocamos ele no Teste interno do Play Console. Com isso ele gera somente um erro com emulador Pixel 2, com Android 9. O erro é esse: 
[logcat.txt](https://github.com/Flutterando/forum/files/14195683/logcat.txt)

Ao pesquisar sobre esse couldn't find "libflutter.so" verifiquei que é um problema que acontece até hoje em dia e vários modelos e versões de Android. E nesses posts existem possíveis soluções que funcionam pra uns casos e outros não. Apliquei várias dessas soluções, porém sem sucesso. Gostaria de alguma ajuda que possa clarear o que pode estar acontecendo.

Flutter doctor:
 Flutter (Channel stable, 2.10.2, on Ubuntu 22.04.3 LTS 6.5.0-15-generic, locale en_US.UTF-8)
    • Flutter version 2.10.2 at /home/guilherme/fvm/versions/2.10.2
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (2 years ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

Lembrando que apesar de estar com a versão do flutter mais antiga (estamos trabalhando na atualização disso), nesses posts, a respeito desse mesmo problema, até mesmo o flutter mais atualizado tem gerado isso.

**Comment by eduardoflorence** [comment]
Dá uma olhada nesses dois comentários para ver se ajuda:
https://github.com/flutter/flutter/issues/32756#issuecomment-868331564
https://github.com/flutter/flutter/issues/83596#issuecomment-1257209669

**Comment by guilhermerp-gh** [comment]
Bom dia, desculpa pela demora no retorno, mas testei com várias dessas soluções e nada até agora.

**Reply by maurojohann** [reply, marked as answer]
Qual arquitetura de processador o pixel 2 está tentando emular?? se for x86, esse é o problema. 
Celulares não usam essa arquitetura, mas os emuladores podem sim tentar emular usando ela. Ai ocorre o erro. Isso só vai acontecer em emulador. Remova esse aparelho da lista de emuladores, pois ele não representa a arquitetura de um dispositivo fisico real. Agora, se mesmo assim quer usar o dispositivo para testar a emulação, defina manualmente as build gerados no apk, removendo da lista x86. Isso não irá impedir de rodar nesse aparelho, mas ele irá emular outra arquitetura pra rodar o software.  
Nessas issue que o @eduardoflorence citou, voce tem a soluçao para esse problema do x86.

**Reply by guilhermerp-gh** [reply]
Humm entendi, muito obrigado por essa informação. A partir de agora não usarei mais essa arquitetura. Vou fazer o ajuste como o @eduardoflorence citou pra não gerar mais esse x86 e subir pra testar nesse teste do google. Assim que tiver notícia volto aqui. Obrigado.

**Reply by guilhermerp-gh** [reply]
Resolvido, muito obrigado ao @eduardoflorence  e ao @maurojohann. Valeu mesmo!
